### PR TITLE
K8sManifest deploy: Fix the container image substitution when the image name contains folders

### LINKED ISSUE #12064: K8sManifest deploy: container substitution failes if containers share a folder
# Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: KubernetesManifest

## Issue Description

When trying to substitute container images that share a common folder only one of the o,ages is substituted.

Deployment spec
```
    spec:
      containers:
      - name: untaggedinsamefoldera
        image: myacr.azurecr.io/folder/image-a
      - name: untaggedinsamefolderb
        image: myacr.azurecr.io/folder/image-b
```

Containers argument
```
myacr.azurecr.io/folder/image-a:1
myacr.azurecr.io/folder/image-b:2
```

Results:
```
    spec:
      containers:
      - name: untaggedinsamefoldera
        image: myacr.azurecr.io/folder/image-a:1
      - name: untaggedinsamefolderb
        image: myacr.azurecr.io/folder/image-b
```

### PR DESCRIPTION
**Pull request by jebbench**
Fixes #12064

### CODE REVIEW

File: Tasks/KubernetesManifestV0/Tests/L0.ts [hunk 1]
```diff
@@ -318,13 +318,16 @@ describe('Kubernetes Manifests Suite', function () {
...
         assert(deploymentObject.spec.template.spec.containers[3].image === 'myacr.azurecr.io/myimage:1', 'untagged image with registry not tagged correctly');
         assert(deploymentObject.spec.template.spec.containers[4].image === 'myimagewithcomment:1', 'untagged image with comment not tagged correctly');
         assert(deploymentObject.spec.template.spec.containers[5].image === 'imagewithhyphen:1', 'manifest regex should work correctly');
-        assert(deploymentObject.spec.template.spec.initContainers[0].image === 'nginx-init:42.1', 'nginx-init image not tagged correctly');
+        assert(deploymentObject.spec.template.spec.containers[6].image === 'myacr.azurecr.io/folder/image-a:1', 'untagged image with common folder not tagged correctly');
+        assert(deploymentObject.spec.template.spec.containers[7].image === 'myacr.azurecr.io/folder/image-b:2', 'untagged image with common folder not tagged correctly');
+
+        assert(deploymentObject.spec.template.spec.initContainers[0].image === 'nginx-init:42.1', 'untagged image with multiple slashes not tagged correctly');
```
Comment (shigupt202) [inline Tasks/KubernetesManifestV0/Tests/L0.ts]: What is the significance of the new message here?

Comment (jebbench) [reply Tasks/KubernetesManifestV0/Tests/L0.ts]: Over zealous copy and pasting - reverted it.

**Comment by shigupt202** [comment]
@jebbench  Please update the task version to 0.164.1

File: Tasks/KubernetesManifestV0/package-lock.json [hunk 2]
```diff
@@ -385,9 +385,9 @@
       }
     },
     "underscore": {
-      "version": "1.9.1",
-      "resolved": "https://registry.npmjs.org/underscore/-/underscore-1.9.1.tgz",
-      "integrity": "sha512-5/4etnCkd9c8gwgowi5/om/mYO5ajCaOgdzj/oW+0eQV9WxKBDZw5+ycmKmeaTXjInS/W0BzpGLo2xR2aBwZdg=="
+      "version": "1.9.2",
+      "resolved": "https://registry.npmjs.org/underscore/-/underscore-1.9.2.tgz",
+      "integrity": "sha512-D39qtimx0c1fI3ya1Lnhk3E9nONswSKhnffBI0gME9C99fYOkNi04xs8K6pePLhvl1frbDemkaBQ5ikWllR2HQ=="
     },
```
Comment (shigupt202) [inline Tasks/KubernetesManifestV0/package-lock.json]: @vithati Does this file need to be checked in?

Comment (vithati) [reply Tasks/KubernetesManifestV0/package-lock.json]: Can check-in

**Comment by vithati** [comment]
LGTM

**Pull request merged**